src/main/java/org/springframework/cloud/deployer/spi/mesos/marathon/MarathonAppDeployer.java

```diff
@@ -112,7 +111,7 @@ public String deploy(AppDeploymentRequest request) {
 
 		app.setCpus(cpus);
 		app.setMem(memory);
-		app.setInstances(Integer.getInteger(request.getDefinition().getProperties().get(COUNT_PROPERTY_KEY)));
+		app.setInstances(Integer.getInteger(request.getDeploymentProperties().get(COUNT_PROPERTY_KEY)));
```

Comment:
Ouch. Found the issue related to count not being passed properly and ... it hurts. `Integer.getInteger()` does not do what you'd think it does.